db/osqlcomm.c

```diff
@@ -5818,42 +5820,64 @@ int osql_process_schemachange(struct ireq *iq, unsigned long long rqid,
     else
         sc->nothrevent = 1;
     sc->finalize = 0;
-    if (sc->original_master_node[0] != 0 &&
-        strcmp(sc->original_master_node, gbl_myhostname))
-        sc->resume = 1;
-
-    iq->sc = sc;
-    sc->iq = iq;
-    sc->is_osql = 1;
-    if (sc->db == NULL) {
-        sc->db = get_dbtable_by_name(sc->tablename);
-    }
-    sc->tran = NULL;
-    if (sc->db)
-        iq->usedb = sc->db;
-
-    if (!timepart_is_timepart(sc->tablename, 1)) {
-        rc = start_schema_change_tran(iq, NULL);
-        if ((rc != SC_ASYNC && rc != SC_COMMIT_PENDING) ||
-            sc->preempted == SC_ACTION_RESUME ||
-            sc->alteronly == SC_ALTER_PENDING) {
-            iq->sc = NULL;
+
+    while(sc){
+        sc = populate_sc_chain(sc);
```

Comment:
Should this be moved above/outside the while loop?

Reply:
no it's important (I think) that it's inside the while loop. It makes it natural for one schema change object to expand into another schema change object that expands. For us specifically, it makes the alters much more natural.